crates/biome_resolver/tests/spec_tests.rs

```diff
@@ -684,3 +684,65 @@ fn test_resolve_alias_with_multiple_target_values() {
         Ok(Utf8PathBuf::from(format!("{base_dir}/src/lib/d.js")))
     );
 }
+
+#[test]
+fn test_resolve_extension_alias() {
+    let base_dir = get_fixtures_path("resolver_cases_8");
+    let fs = OsFileSystem::new(base_dir.clone());
+
+    let options = ResolveOptions {
+        default_files: &["index"],
+        extensions: &["js"],
+        extension_alias: &[("js", &["ts", "js"]), ("mjs", &["mts"])],
+        ..Default::default()
+    };
+
+    assert_eq!(
+        resolve("./index.js", &base_dir, &fs, &options),
+        Ok(Utf8PathBuf::from(format!("{base_dir}/index.ts"))),
+        "should alias fully specified file",
+    );
+
+    assert_eq!(
+        resolve("./dir/index.js", &base_dir, &fs, &options),
+        Ok(Utf8PathBuf::from(format!("{base_dir}/dir/index.ts"))),
+        "should alias fully specified file when there are two alternatives",
```

Comment:
The message seems a bit misleading, because there are no two alternatives in `dir/`. Maybe this doesn't test what you expect it to be testing?